### ISSUE TITLE
Invalid Island Image Names

### ISSUE THREAD

**Issue by SubstituteR** [GrapeshotGames/ServerGridEditor#45]
There are several island images who's file extension suggests they are JPG images, but they're really PNG images.  
As-Is this doesn't break the editor, but tools to process these images will fail.
They should be changed to PNG to reflect their contents correctly.  Doing so will ensure both consistency and prevent future issues where the file extension might matter.  (Image Magick for one treats the file extension as the file type.)

**Comment by smooreace** [comment]
My exports are JPGs, and are not renamed PNGs.

**Comment by rezalas** [comment]
@smooreace Sub isn't saying the exports are PNGs, but talking about the actual island images that are part of the editor.

**Comment by smooreace** [comment]
My bad... that makes sense now that you said that.

**Comment by jwvanderbeck** [comment]
Just found this as well when trying to do batch operations with Photoshop.  Many island images have the jpg extension when they are in fact png images.

**Comment by jwvanderbeck** [comment]
DO NOT try renaming them either.  I just tried that and had to deal with about 500 error boxes upon starting the application :D  Seems the images are hardcoded or some such nonsense.

**Comment by duxck** [comment]
Are we talking about island images stored in \IslandImages ?
Not sure if I understood this correctly. But I renamed all jpgs to png with ren *.jpg *.png and was able to export everything fine on the 2x2, 4x4 and 15x15.

Worked fine to export all, export map image, export cell image and export slippy map.

**Comment by jwvanderbeck** [comment]
Yes the \IslandImages,but it has nothing to do with the export functions. I'm curious how you even opened the program though after renaming them because wjhen I did that it just threw up  fatal error box for each and every JPG that I had renamed to PNG saying it was missing the image.

**Comment by smooreace** [comment]
Why would you rename them? If you're wanting to edit the images then you
shouldn't be screwing with the original files anyway. Thats something we
are working on to make them look proper on an old paper looking background.

**Comment by duxck** [comment]
Okay, my bad. I renamed in the wrong project folder.

The function that calls the error message is in Islands.cs 

```
public Image GetImage(bool optimized = false)
{
	if (cachedImg == null)
	{
		if (File.Exists(imagePath))
			cachedImg = Image.FromFile(imagePath);
		else
		{
			MessageBox.Show("Could not find image (Setting to blank image):" + imagePath, "Missing Image Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
			Bitmap bmp = new Bitmap(78, 78);
			cachedImg = Image.FromHbitmap(bmp.GetHbitmap());
		}
	}
	if(optimized)
	{
		if (cachedOptimizedImg == null)
		{
			if (File.Exists(imagePath))
			{
				cachedOptimizedImg = Image.FromFile(imagePath);
				Bitmap bmp = new Bitmap(cachedOptimizedImg, cachedOptimizedImg.Width / 16, cachedOptimizedImg.Height / 16);
				cachedOptimizedImg = (Image)bmp;
			}
		}
		if (cachedOptimizedImg != null)
			return cachedOptimizedImg;
	}
	return cachedImg;
}
```

imagePath is set by the constructor in Island,cs 

```
public Island(string name, float x, float y, string imagePath, int landscapeMaterialOverride
            , List<string> sublevelNames, Dictionary<string, string> spawnerOverrides, float minTreasureQuality, float maxTreasureQuality, bool useNpcVolumesForTreasures,
            bool useLevelBoundsForTreasures, bool prioritizeVolumesForTreasures, string IslandTreasureBottleSupplyCrateOverrides, List<string> extraSublevels)
        {
            this.name = name;
            this.x = x;
            this.y = y;
            this.imagePath = imagePath;
            this.landscapeMaterialOverride = landscapeMaterialOverride;
            this.sublevelNames = sublevelNames;
            this.spawnerOverrides = spawnerOverrides;
            this.minTreasureQuality = minTreasureQuality;
            this.maxTreasureQuality = maxTreasureQuality;
            this.useNpcVolumesForTreasures = useNpcVolumesForTreasures;
            this.useLevelBoundsForTreasures = useLevelBoundsForTreasures;
            this.prioritizeVolumesForTreasures = prioritizeVolumesForTreasures;
            this.islandTreasureBottleSupplyCrateOverrides = IslandTreasureBottleSupplyCrateOverrides;
            this.extraSublevels = extraSublevels;
        }
```
I wasn't able to identify where the island is instantiated though. So I'm stuck.

**Comment by jwvanderbeck** [comment]
> Why would you rename them? If you're wanting to edit the images then you shouldn't be screwing with the original files anyway. Thats something we are working on to make them look proper on an old paper looking background.

Renamed them because they are not JPG files, they are PNG files.  Image editor applications, like Photoshop will fail to open them because it sees a JPG extension and tries to open them as JPG.  In my case I was trying to batch process all the images and this meant that all the fake JPGs would not be processed.  I had to rename them to PNG, do the processing, then rename them back to JPG so the SGE wouldn't error out. Whic was a bit of a pain :)

Bottom line, tl;dr is that they should just be named properly with the proper extension in the first place.  I can't think of one reason why they need to be given the wrong extension other than the SGE is looking for them that way.

**Comment by TheShak64** [comment]
At first i thought the same as SubstituteCS where PNG files are not recognized by the ServerGridEditor. However some of the PNG files do show up in the list of available islands. Some are not available on the list on the ServerGridEditor. Is this because the ServerGridEditor is not updated?
